201-vm-msi-linux-terraform/azuredeploy.json

```diff
@@ -0,0 +1,320 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for the Virtual Machine."
+      }
+    },
+    "authenticationType": {
+      "type": "string",
+      "defaultValue": "password",
+      "allowedValues": [
+        "password",
+        "sshPublicKey"
+      ],
+      "metadata": {
+        "description": "Authentication type"
+      }
+    },
+    "adminPasswordorSSHKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "OS Admin password or SSH Key depending on value of authentication type"
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "metadata": {
+        "description": "The size of the VM to create"
+      },
+      "defaultValue": "Standard_D1_V2"
+    },
+    "desktopInstall": {
+      "type": "bool",
+      "defaultValue": false,
+      "metadata": {
+        "description": "Installs Ubuntu Mate desktop GUI"
+      }
+
+    }
+  },
+  "variables": {
+    "dnsLabelPrefix": "[concat('msi',uniquestring(resourceGroup().id,deployment().name))]",
+    "_artifactsLocation": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-msi-linux-terraform",
+    "infraStorageAccountName": "[take(concat('storeinfra', uniquestring(resourceGroup().id), variables('dnsLabelPrefix')),24)]",
+    "stateStorageAccountName": "[take(concat('storestate', uniquestring(resourceGroup().id), variables('dnsLabelPrefix')),24)]",
+    "nicName": "[concat('nic',uniquestring(resourceGroup().id,deployment().name))]",
+    "networkSecurityGroupName": "[concat('nsg',uniquestring(resourceGroup().id,deployment().name))]",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "Subnet",
+    "subnetPrefix": "10.0.0.0/24",
+    "publicIPAddressName": "[concat('pip',uniquestring(resourceGroup().id,deployment().name))]",
+    "vmName": "[concat('vm',uniquestring(resourceGroup().id,deployment().name))]",
+    "virtualNetworkName": "[concat('vnet',uniquestring(resourceGroup().id,deployment().name))]",
+    "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]",
+    "linuxConfigurationChosen": "[concat('linuxConfiguration', parameters('authenticationType'))]",
```

Comment:
simplify this section around ssh or pw authn - see: https://github.com/Azure/azure-quickstart-templates/blob/master/201-vm-new-or-existing-conditions/azuredeploy.json

Reply:
Done